tests for Sprint-3

Taler777 opened the pull request:
tests for Sprint-3
**Для преподавателей:**
- было бы неплохо в HrDepartment.processNextRequest() делать проверку incomeBox на null перез вызовом incomeBox.poll(), потому что на следующей строке можно словить NPE

Commit:
test for Sprint-3

shariktlt-devops reviewed (approved):
:ok: build ok, see: https://jenkins.shariktlt.ru/job/kotlin_school_2/job/PR_BUILDER/457/

Commit:
oop for Sprint-3

shariktlt-devops reviewed (changes requested):
:no_entry: build failed, see: https://jenkins.shariktlt.ru/job/kotlin_school_2/job/PR_BUILDER/520/

Commit:
fix clock for Sprint-3

shariktlt-devops reviewed (approved):
:ok: build ok, see: https://jenkins.shariktlt.ru/job/kotlin_school_2/job/PR_BUILDER/523/

SberKotlinSchool commented on qa/src/test/kotlin/ru/sber/qa/HrDepartmentTest.kt:
```diff
@@ -0,0 +1,119 @@
+        HrDepartment.receiveRequest(certificateRequest)
+
+        //then
+        HrDepartment.javaClass.getDeclaredField("incomeBox").let {
```
дублирование кода - можно вынести в отдельный метод

Taler777 replied:
Исправил

SberKotlinSchool commented on qa/src/test/kotlin/ru/sber/qa/HrDepartmentTest.kt:
```diff
@@ -0,0 +1,119 @@
+            it.isAccessible = true
+            val outcomeOutcomeList = it.get(outcomeOutcome) as LinkedList<*>
+
+            HrDepartment.processNextRequest(0L)
```
вызов метода и ассерт следует вынести за let

Taler777 replied:
Исправил

Commit:
Sprint-3
- исправление замечаний код-ревью по "Тестированию"
- IO архивирование, разархивирование

shariktlt-devops reviewed (approved):
:ok: build ok, see: https://jenkins.shariktlt.ru/job/kotlin_school_2/job/PR_BUILDER/630/

SberKotlinSchool commented on oop/src/main/kotlin/ru/sber/oop/DataClasses.kt:
```diff
@@ -2,10 +2,24 @@ package ru.sber.oop
+        if (other == null) return false
+        if (this === other) return true
+        if (other !is User) return false
+        if (this.name == other.name && this.age == other.age && this.city == other.city) {
```
тут будет ошибка если поле city не инициализировано

Taler777 replied:
Не соглашусь:
By convention, an expression like a == b is translated to:

a?.equals(b) ?: (b === null)
If a is not null, it calls the equals(Any?) function, otherwise (a is null) it checks that b is referentially equal to null.

SberKotlinSchool commented on io/src/main/kotlin/ru/sber/io/Archivator.kt:
```diff
@@ -1,24 +1,64 @@
      * Извлечь из архива logfile.zip файл и сохарнить его в том же каталоге с именем unzippedLogfile.log
      */
     fun unzipLogfile() {
-
```
unzippedLogfile.log пустой получается

Taler777 replied:
Исправил

Commit:
Sprint-3
- исправление замечаний код-ревью по IO архивирование, разархивирование

shariktlt-devops reviewed (approved):
:ok: build ok, see: https://jenkins.shariktlt.ru/job/kotlin_school_2/job/PR_BUILDER/699/

Commit:
Sprint-3
NIO

shariktlt-devops reviewed (approved):
:ok: build ok, see: https://jenkins.shariktlt.ru/job/kotlin_school_2/job/PR_BUILDER/729/
